### ISSUE TITLE
Improve horizontal layout usage of blocks

### ISSUE THREAD

**Issue by dinhtungdu** [woothemes/woocommerce#44759]
As @nerrad mentioned:

> The default behaviour for the blocks in a horizontal layout (via usage of Row block – image on left) is not great and cleaning things up by removing the heading block and changing the justification still doesn’t make things much better (see image on right).

> There definitely needs to be some better layout behaviour for horizontal layouts and it’d be nice to have options for things like the price block to fill the content.

| | |
|--------|--------|
| ![image](https://github.com/woocommerce/woocommerce/assets/5423135/33d5b8d7-302d-477d-9430-17b9d28d48b5) | ![image](https://github.com/woocommerce/woocommerce/assets/5423135/9bec5b02-6617-4c44-935e-228c109c75fa) |

**Comment by dinhtungdu** [comment]
We can use the HM site as a good example of a horizontal filter layout:
![image](https://github.com/woocommerce/woocommerce/assets/5423135/5d324095-1ad6-4c6c-b0d8-91d220f72823)

IMO, we can handle this layout like the following:
- Update the `Product Filter` block to have multiple layout styles.
- The default style can be called `Vertical`.
- Add an option to the layout style called `Horizontal` or `Dropdown`, which takes the new filter blocks as the inner block. It shows a customizable title by default and when shoppers click on the title, it shows the filter block.
- We can (and should) utilize the [Block Styles API](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-styles/).
- We need additional patterns for horizontal layouts.
- We can also create additional variations of the `Product Filter` to let merchants add the horizontal filter block through the inserter.

**Comment by nerrad** [comment]
Let's make sure we get some design input on this from @jarekmorawski.

> We can also create additional variations of the Product Filter to let merchants add the horizontal filter block through the inserter.

Tell me more. I'm having trouble reconciling this with the Block Styles variations idea.

**Comment by dinhtungdu** [comment]
> > We can also create additional variations of the Product Filter to let merchants add the horizontal filter block through the inserter.
> 
> Tell me more. I'm having trouble reconciling this with the Block Styles variations idea.

Both new and current filter blocks use the wrapper variation to insert the block to the page from the inserter. For example, we have `Filter by Price` in the inserter is the variation of the price filter block. What I mean is creating another set of variations for horizontal options, like `Filter by Price - Horizontal`.

Instead of Styles API, a `SelectControl` with an extra attribute will work. But I think we should first try utilizing the official API to see if we can make the style extensible. Besides, the style selector with previews is a better UX.

**Comment by kkoteen** [comment]
Hi @dinhtungdu - a lot has changed with the filter block since this issue was opened, and horizontal filters would still be a nice addition. 

Any updates you want to add before we make a call on this one?

**Comment by dinhtungdu** [comment]
@kkoteen thanks for the ping. For now, I don't think we should follow this


> * Update the `Product Filter` block to have multiple layout styles.
> * The default style can be called `Vertical`.
> * Add an option to the layout style called `Horizontal` or `Dropdown`, which takes the new filter blocks as the inner block. It shows a customizable title by default and when shoppers click on the title, it shows the filter block.

Instead we should provide a primitive which have the trigger and a dialog that can render the whole filter block inside. And we provide this configuration through pattern, not block setting.

**Comment by keoshi** [comment]
Hi, folks. Jumping on this one from the design side of things.

I understand the original issue, but agree with @kkoteen (hi, Katie!) that a lot has changed since this was initially filed. For example, every attribute now has a corresponding filter and people can compose their view in as simple or as complex way as they want to. Here's an example of what a more complex composition would mean for the layout:

<img width="1425" height="821" alt="Image" src="https://github.com/user-attachments/assets/100230e9-730c-4977-b9be-ff9b3531333f" />

Regarding the HM example, that still feels pertinent as I could see Woo offering a way for filters to always show up as individual dropdown groups. One avenue that seems worth exploring would be to replicate the architecture of the Navigation block in the Editor and let people add filters as sub/child items.

<img width="740" height="352" alt="Image" src="https://github.com/user-attachments/assets/0c984b4c-23bb-453e-8b65-a36efed7f538" />

Even see this as something that could be contributed upstream to Core as a new block; i.e.: “Dropdown”.

Would love to hear your thoughts.

**Comment by dinhtungdu** [comment]
> Even see this as something that could be contributed upstream to Core as a new block; i.e.: “Dropdown”.

I agree we should provide block like this instead of block settings. We already have some like Details, Accordion.
